src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -413,64 +447,77 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
+			} else {
+				self?.registerTestAndGetResult(with: testQrCodeInformation, submissionConsentGiven: true, isLoading: isLoading)
+			}
+		}
+
+		if let testInformation = testInformation {
+			testInformationSuccess(testInformation)
+		} else {
+			let scannerViewController = ExposureSubmissionQRScannerViewController(
+				onSuccess: { [weak self] testQrCodeInformation in
+					self?.presentedViewController?.dismiss(animated: true) {
+						testInformationSuccess(testQrCodeInformation)
 					}
-				case .codeNotFound:
-					DispatchQueue.main.async {
-						let alert = UIAlertController.errorAlert(
-							title: AppStrings.ExposureSubmissionError.qrAlreadyUsedTitle,
-							message: AppStrings.ExposureSubmissionError.qrAlreadyUsed,
-							okTitle: AppStrings.Common.alertActionCancel,
-							secondaryActionTitle: AppStrings.Common.alertActionRetry,
-							completion: { [weak self] in
+				},
+				onError: { [weak self] error, reactivateScanning in
+					switch error {
+					case .cameraPermissionDenied:
+						DispatchQueue.main.async {
+							let alert = UIAlertController.errorAlert(message: error.localizedDescription, completion: {
 								self?.presentedViewController?.dismiss(animated: true)
-							},
-							secondaryActionCompletion: { reactivateScanning() }
-						)
-						self?.presentedViewController?.present(alert, animated: true)
+							})
+							self?.presentedViewController?.present(alert, animated: true)
+						}
+					case .codeNotFound:
+						DispatchQueue.main.async {
+							let alert = UIAlertController.errorAlert(
+								title: AppStrings.ExposureSubmissionError.qrAlreadyUsedTitle,
+								message: AppStrings.ExposureSubmissionError.qrAlreadyUsed,
+								okTitle: AppStrings.Common.alertActionCancel,
+								secondaryActionTitle: AppStrings.Common.alertActionRetry,
+								completion: { [weak self] in
+									self?.presentedViewController?.dismiss(animated: true)
+								},
+								secondaryActionCompletion: { reactivateScanning() }
+							)
+							self?.presentedViewController?.present(alert, animated: true)
+						}
+					case .other:
+						Log.error("QRScannerError.other occurred.", log: .ui)
 					}
-				case .other:
-					Log.error("QRScannerError.other occurred.", log: .ui)
+				},
+				onCancel: { [weak self] in
+					self?.presentedViewController?.dismiss(animated: true)
 				}
-			},
-			onCancel: { [weak self] in
-				self?.presentedViewController?.dismiss(animated: true)
-			}
-		)
+			)
 
-		let qrScannerNavigationController = UINavigationController(rootViewController: scannerViewController)
-		qrScannerNavigationController.modalPresentationStyle = .fullScreen
+			let qrScannerNavigationController = UINavigationController(rootViewController: scannerViewController)
+			qrScannerNavigationController.modalPresentationStyle = .fullScreen
+
+			navigationController?.present(qrScannerNavigationController, animated: true)
```

Comment:
Is it Ok, that there can be a call to "self?.showOverrideTestNotice(...)" and after that a call to "navigationController?.present(...)" ?

Seems that there can be 2 navigation calls after each other?